Dockerfile

```diff
@@ -1,24 +1,41 @@
-# Stage 1: Build the application
-FROM node:alpine AS builder
+##############
+# Base Image #
+##############
+FROM node:20.12.1-alpine3.19 AS builder
 
 WORKDIR /app
 
 COPY package.json pnpm-lock.yaml ./
 
 RUN npm install -g pnpm && pnpm install
 
+#####################
+# Development Image #
+#####################
+FROM builder AS development
+
+WORKDIR /app
+
 COPY . .
 
-# Stage 2: Create a smaller production image
-FROM alpine:3.14
+EXPOSE 3000
+
+CMD ["pnpm", "dev"]
```

Comment:
if we are running `pnpm build` and `pnpm start` command later,  we don't need this, neither the `EXPOSE 3000` here